lib/recaptcha/http/mock_http_client.ex

```diff
@@ -24,14 +23,21 @@ defmodule Recaptcha.Http.MockClient do
   end
 
   def request_verification(
-        "response=invalid_response&secret=6LeIxAcTAAAAAGG-vFI1TnRWxMZNFuojJ4WifJWe" =
-          body,
+        "response=invalid_response&secret=6LeIxAcTAAAAAGG-vFI1TnRWxMZNFuojJ4WifJWe" = body,
         options
       ) do
     send(self(), {:request_verification, body, options})
     {:error, [:"invalid-input-response"]}
   end
 
+  # This clause is used to simulate reCAPTCHA's API returning multiple error codes as the old test
+  # wasn't working anymore and I was not able to make the API trigger multiple error codes manually,
+  # even though the `error-codes` field in the response is a list of errors.
+  def request_verification("response=not_valid&secret=test_secret" = body, options) do
+    send(self(), {:request_verification, body, options})
+    {:error, [:invalid_input_response, :invalid_input_secret]}
+  end
```

Comment:
Maybe it's worth changing the tests to use [bypass](https://hex.pm/packages/bypass) at some point 🤔

Reply:
@vorce Thank you so much for that recommendation! I had never used bypass before and it was awesome to get the chance to learn about it and use it 🙌 

I have pushed [this commit](https://github.com/remoteoss/recaptcha/pull/1/commits/d1b9527ee0c7817fb6eaa83814f3b615db0e2769) which introduces bypass a project dependency and uses it in the tests, please give it a review as it's my first time using it.

I believe it's also worth noting that I didn't delete the `Recaptcha.Http.MockClient` module as the tests are using the `assert_received` call in order to validate the arguments provided to the `request_verification/2` function. I did update it so that it only sends the message, with `send/2`, but then calls the real HTTP client. Do you think it's worth it to introduce Mox in order to perform this argument validation and completely remove this module? 🤔

Reply:
Wow, nice one @joaofcosta ! Looks great 👍 

> Do you think it's worth it to introduce Mox in order to perform this argument validation and completely remove this module?

Nah, unless it bothers you hehe. I think what we got now is fine!